exporter/containerimage/writer.go

```diff
@@ -299,10 +299,18 @@ func (ic *ImageWriter) extractAttestations(ctx context.Context, s session.Group,
 				for _, subject := range att.Subjects {
 					switch subject2 := subject.(type) {
 					case *result.InTotoSubjectSelf:
-						statements[i].Subject = append(statements[i].Subject, intoto.Subject{
-							Name:   "_",
-							Digest: result.DigestToDigestMap(desc.Digest),
-						})
+						var names []string
+						if opts.ImageName == "" {
+							names = []string{"_"}
+						} else {
+							names = strings.Split(opts.ImageName, ",")
+						}
+						for _, name := range names {
+							statements[i].Subject = append(statements[i].Subject, intoto.Subject{
+								Name:   name,
```

Comment:
Should this be converted to https://github.com/package-url/purl-spec ?

Reply:
Maybe. I'm not entirely sure, in-toto doesn't require it, so it's up to us. I do think we want to make sure that the platform is present in the name though (to ensure uniqueness), since then we'll have a 1-to-1 relationship between digests and names.

So maybe PURL would be the right choice here, with the version field containing the image tag, and the image repository location and architecture/os data in the qualifiers? Not sure whether we should use the `sha256:` in the name though, since that would be duplicated.